lib/features/mailbox/domain/usecases/get_all_mailbox_interactor.dart

```diff
@@ -5,21 +5,30 @@ import 'package:jmap_dart_client/jmap/core/properties/properties.dart';
 import 'package:tmail_ui_user/features/mailbox/domain/repository/mailbox_repository.dart';
 import 'package:tmail_ui_user/features/mailbox/domain/state/get_all_mailboxes_state.dart';
 import 'package:tmail_ui_user/features/mailbox/domain/extensions/list_mailbox_extension.dart';
-import 'package:tmail_ui_user/features/mailbox/domain/extensions/mailbox_extension.dart';
+import 'package:model/model.dart';
 
 class GetAllMailboxInteractor {
-  final MailboxRepository mailboxRepository;
+  final MailboxRepository _mailboxRepository;
 
-  GetAllMailboxInteractor(this.mailboxRepository);
+  GetAllMailboxInteractor(this._mailboxRepository);
 
   Stream<Either<Failure, Success>> execute(AccountId accountId, {Properties? properties}) async* {
     try {
       yield Right<Failure, Success>(LoadingState());
-      final mailboxList = await mailboxRepository.getAllMailbox(accountId, properties: properties);
-      final tupleList = mailboxList.splitMailboxList((mailbox) => mailbox.hasRole());
-      yield Right<Failure, Success>(GetAllMailboxSuccess(defaultMailboxList: tupleList.value1, folderMailboxList: tupleList.value2));
+
+      final streamListMailbox = _mailboxRepository.getAllMailbox(accountId, properties: properties);
+
+      final streamResult = streamListMailbox.asyncExpand((response) async* {
+        final tupleList = response.mailboxes?.splitMailboxList((mailbox) => mailbox.hasRole()) ?? Tuple2([], []);
+        yield Right<Failure, Success>(GetAllMailboxSuccess(
+          defaultMailboxList: tupleList.value1,
+          folderMailboxList: tupleList.value2,
+          currentState: response.state));
+      });
```

Comment:
How about using `async` package to make it more readable?

Reply:
I find it pretty obvious here. Please describe your idea

Reply:
please discuss this in https://github.com/linagora/tmail-flutter/pull/137, I re-wrote it with the test.

Reply:
Ok. It works well.